### PR TITLE
E2EE dialog crash on clicking "Set as Encrypted" multiple times

### PR DESCRIPTION
**Pull request by surinder-tsys**
**Steps to reproduce:**

1. Click on Set as encrypted option to do the encryption
2. Once the dialog shows do back press to remove the dialog
3. Repeat 1st and 2nd step this will lead to crash.

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by ZetaTom** [comment]
> **Steps to reproduce:**
> 
>   1. Click on Set as encrypted option to do the encryption
>   2. Once the dialog shows do back press to remove the dialog
>   3. Repeat 1st and 2nd step this will lead to crash.

Using the current master branch (6bbd629), I'm unfortunately not able to observe this behaviour.

**Comment by ZetaTom** [comment]
@surinder-tsys, could you please provide us with your configuration (device model, manufacturer, Android and app version) and check if you are still able to reproduce this behaviour? There have been some changes (#11694) made to this dialogue since this pull request was opened.

**Comment by surinder-tsys** [comment]
@ZetaTom The crash is still reproducible and the other changes has no effect to it.
Its trickier to reproduce. Its not device specific as the crash is happening due to null Context.

To reproduce it --> On clicking "Set as encrypted" option do the back press(as quickly you can) before the api call finishes. 

Since AsyncTask is used for api call and if the dialog is not visible it will lead to crash in onPostExecute method. So using WeakReference for Context will fix this issue.

Refer screenshot: 
E2ee crash

File: app/src/main/java/com/owncloud/android/ui/dialog/SetupEncryptionDialogFragment.java [hunk 1]
```diff
@@ -303,47 +313,58 @@ protected void onPreExecute() {
 
         @Override
         protected String doInBackground(Void... voids) {
+            Context context = mWeakContext.get();
             // fetch private/public key
             // if available
             //  - store public key
             //  - decrypt private key, store unencrypted private key in database
```
Comment (ZetaTom) [inline app/src/main/java/com/owncloud/android/ui/dialog/SetupEncryptionDialogFragment.java:321]:
```suggestion
            // fetch private/public key
            // if available
            //  - store public key
            //  - decrypt private key, store unencrypted private key in database

            Context context = mWeakContext.get();
```

**Review by ZetaTom — COMMENTED** [review]
@surinder-tsys, thank you for confirming the details.

After artificially increasing the network latency, I was able to reliably reproduce this issue and can now confirm that this change fixes the problem and works as expected.

File: app/src/main/java/com/owncloud/android/ui/dialog/SetupEncryptionDialogFragment.java (hunk 1 above)
Comment (AndyScherzinger) [reply app/src/main/java/com/owncloud/android/ui/dialog/SetupEncryptionDialogFragment.java:321]: Sry @ZetaTom I already merged a second ago. can you create a separate PR with the polishing, I can review and merge that one then. Sorry for the extra work.

Comment (ZetaTom) [reply app/src/main/java/com/owncloud/android/ui/dialog/SetupEncryptionDialogFragment.java:321]: No problem, I've opened a new pull request: #11758

**Pull request merged**